### ISSUE TITLE
feature request: passing number instead of relation object should default to handling it as id

### ISSUE THREAD

**Issue by NoNameProvided** [typeorm/typeorm#536]
When I would like to save an entity I have to specify an object with it's relation, but as far as I know TypeORM currently use this to get the primary key of the relation. 

My question, can we handle numbers as PKs when they are provided instead the related entity?

An example:

```ts
@Entity()
export class Photo {

    @PrimaryGeneratedColumn()
    id: number;

    @OneToOne(type => PhotoMetadata, photoMetadata => photoMetadata.photo)
    metadata: PhotoMetadata;
}

@Entity()
export class PhotoMetadata {

    @PrimaryGeneratedColumn()
    id: number;

    @OneToOne(type => Photo)
    @JoinColumn()
    photo: Photo;
}

// later in app
let validPhotoId = 5;
let metadata = new PhotoMetadata();
metadata.photo = 5; // instead of metadata.photo = photo; 

await metadataRepository.persist(metadata);
```

**Comment by NoNameProvided** [comment]
I know we have the following workaround for now:

```ts
metadata.photo = { id:5 };
```

But allowing to pass the number at the first place would make the usage easier. (Eg, lot of the times only ids coming  from the frontend and we have to do ids.map(id => { return { id } }))

**Comment by pleerock** [comment]
there is kinda hacky (but official) way of doing this:

```ts

@Entity()
export class PhotoMetadata {

    @PrimaryGeneratedColumn()
    id: number;

    @Column({ nullable: true, type: "int" })
    photoId: number;

    @OneToOne(type => Photo)
    @JoinColumn({ name: "photoId" })
    photo: Photo;
}
```

Then you can do:

```ts
let metadata = new PhotoMetadata();
metadata.photoId = 5;
```

But probably it would be very nice if we allow for users to do `metadata.photo = 5;` as well

**Comment by NoNameProvided** [comment]
How does this work when let's say photo is an array of `Photos`?

**Comment by pleerock** [comment]
it wont work this way

**Comment by felipesabino** [comment]
@pleerock Should this work on `ManyToOne` relations? Just tried here and am getting the following error

```
 error: column "photoId" specified more than once error: column "photoId" specified more than once
```

my entity is 

```
@Entity()
export class PhotoMetadata {

    @PrimaryGeneratedColumn()
    id: number;

    @Column({ nullable: true, type: "int" })
    photoId: number;

    @ManyToOne(type => Photo, {
      cascadeInsert: true,
      onDelete: 'CASCADE',
     })
    @JoinColumn({ name: "photoId" })
    photo: Photo;
}```

**Comment by apologetics** [comment]
@felipesabino You have both the column and the join column having the same name in the database. For the `photoId` which is a nember and a separate column, naming strategy applies the name of the variable as name was not provided in decorator options. As for the join column, you forcely set the name of the column with the 'name' param.

You need to either rename the JoinColumn column name,  or pass genuine name to photoId decorator. E.g.:

```
@Column ({nullable: true, type:'int', name:'parent'})
photoId: number;

// OR

@JoinColumn ({ name: 'photo_id' })
photo: Photo;
```

**Comment by felipesabino** [comment]
@apologetics Sorry, I don't think I got it.

> You have both the column and the join column having the same name in the database.

I guess that is exactly the point, right? I only need one column for the relation (aka foreign key) on my table, but have access to the foreign key id without having to perform any join.

The goal is to receive an instance of `PhotoMetadata` from a single query and being able to know what it the `id` of the `photo` related to that entity without needing to go to the database.

(BTW, the original request from _NoNameProvided _ asked on a way to save this information without having to go to the database and fetching extra data, which is kind of similar in the current context)

So in the end I would have my entity class with both `photo: Photo` and `photoId: number`, but in the database I would have just one column that it the one used to build the relation

**Comment by apologetics** [comment]
Then what is the point of marking `photoId: number` as a `@Column` if you do not want it to exist in the database?

**Comment by pleerock** [comment]
>  error: column "photoId" specified more than once error: column "photoId" specified more than once

you should not have this error. Try to remove output directory/output js files and try again. Log sql queries and show what it executes

**Comment by felipesabino** [comment]
@pleerock 

Just as an FYI the SQL log shows it actually duplicating the `photoId` column when it creates the column but I could only reproduce it on `0.0.11` (version being used by my project), not on `master`.

**Comment by pleerock** [comment]
no, it should not. Try on latest version and report a new issue if you have this problem

**Comment by pleerock** [comment]
This is implemented in `0.2.0-alpha` versions. Please try it in a new version (npm i typeorm@next).

**Comment by WojtylaCZ** [comment]
Is this resolved? Is there a no-hack way to do :
```
let validPhotoId = 5;
let metadata = new PhotoMetadata();
metadata.photo = 5; // instead of metadata.photo = photo; 
```
